### PR TITLE
Add example of marking Vault values as sensitive variables

### PR DESCRIPTION
**Pull request by nywilken**
Related to: https://github.com/hashicorp/packer/issues/11465

### CODE REVIEW

File: website/content/docs/templates/hcl_templates/functions/contextual/vault.mdx [hunk 1]
```diff
@@ -41,7 +41,24 @@ locals {
...
+If the Vault secret contains a highly sensitive value the `local` block, not to be confused with
+the `locals` block, can be used to mark the value as sensitive.
+
+```hcl
+local "foo" {
+    expression = vault("secrets/hello", "foo")
+    sensitive  = true
+}
+
+The `local` block example accesses the Vault path `secret/foo` and returns the value
```
Comment (aram535) [inline website/content/docs/templates/hcl_templates/functions/contextual/vault.mdx:53]: secrets

Comment (nywilken) [inline website/content/docs/templates/hcl_templates/functions/contextual/vault.mdx:53]:
```suggestion
The `local` block example accesses the Vault path `secrets/foo` and returns the value
```

Comment (nywilken) [reply website/content/docs/templates/hcl_templates/functions/contextual/vault.mdx]: Thanks for the review @aram535

**Review by azr — APPROVED** [review]
Nice !

**Pull request merged**